seatunnel-e2e/seatunnel-connector-v2-e2e/connector-iceberg-e2e/src/test/java/org/apache/seatunnel/e2e/connector/iceberg/IcebergSinkWithBranchIT.java

```diff
@@ -0,0 +1,184 @@
+        type = {},
+        disabledReason = "")
+@DisabledOnOs(OS.WINDOWS)
+public class IcebergSinkWithBranchIT extends TestSuiteBase {
+
+    private static final String CATALOG_DIR = "/tmp/seatunnel/iceberg/hadoop-sink/";
+
+    private static final String NAMESPACE = "seatunnel_namespace";
+
+    private String zstdUrl() {
+        return "https://repo1.maven.org/maven2/com/github/luben/zstd-jni/1.5.5-5/zstd-jni-1.5.5-5.jar";
+    }
+
+    @TestContainerExtension
+    protected final ContainerExtendedFactory extendedFactory =
+            container -> {
+                container.execInContainer("sh", "-c", "mkdir -p " + CATALOG_DIR);
+                container.execInContainer("sh", "-c", "chmod -R 777  " + CATALOG_DIR);
+                container.execInContainer(
+                        "sh",
+                        "-c",
+                        "mkdir -p /tmp/seatunnel/plugins/Iceberg/lib && cd /tmp/seatunnel/plugins/Iceberg/lib && wget "
+                                + zstdUrl());
+            };
+
+    private final String NAMESPACE_TAR = NAMESPACE + ".tar.gz";
+    protected final ContainerExtendedFactory containerExtendedFactory =
+            new ContainerExtendedFactory() {
+                @Override
+                public void extend(GenericContainer<?> container)
+                        throws IOException, InterruptedException {
+                    FileUtils.createNewDir(CATALOG_DIR);
+                    container.execInContainer(
+                            "sh",
+                            "-c",
+                            "cd "
+                                    + CATALOG_DIR
+                                    + " && tar -czvf "
+                                    + NAMESPACE_TAR
+                                    + " "
+                                    + NAMESPACE);
+                    container.copyFileFromContainer(
+                            CATALOG_DIR + NAMESPACE_TAR, CATALOG_DIR + NAMESPACE_TAR);
+                    extractFiles();
+                }
+
+                private void extractFiles() {
+                    ProcessBuilder processBuilder = new ProcessBuilder();
+                    processBuilder.command(
+                            "sh", "-c", "cd " + CATALOG_DIR + " && tar -zxvf " + NAMESPACE_TAR);
+                    try {
+                        Process process = processBuilder.start();
+                        int exitCode = process.waitFor();
+                        if (exitCode == 0) {
+                            log.info("Extract files successful.");
+                        } else {
+                            log.error("Extract files failed with exit code " + exitCode);
+                        }
+                    } catch (IOException | InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                }
+            };
+
+    @TestTemplate
+    public void testInsertAndCheckDataE2e(TestContainer container)
```

Comment:
How to validate `iceberg.table.commit-branch="commit-branch"`

Reply:
> How to validate `iceberg.table.commit-branch="commit-branch"`

'loadDataFromIcebergTableBranch()  uses table.newScan().useRef(commitBranch) to specify that data should be read from a branch.